api/napi/__test__/js_value_conversion.spec.ts

```diff
@@ -0,0 +1,307 @@
+
+test('get/set string properties', (t) => {
+
+  let compiler = new ComponentCompiler;
+  let definition = compiler.buildFromSource(`export component App { in-out property <string> name: "Initial"; }`, "");
+  t.not(definition, null);
+
+  let instance = definition!.create();
+  t.not(instance, null);
+
+  t.is(instance!.getProperty("name"), "Initial");
+
+  instance!.setProperty("name", "Hello");
+  t.is(instance!.getProperty("name"), "Hello");
+
+  t.throws(() => {
+    instance!.setProperty("name", 42)
+  },
+    {
+      code: "InvalidArg",
+      message: "expect String, got: Number"
+    }
+  );
+
+  t.throws(() => {
+    instance!.setProperty("name", { "blah": "foo" })
+  },
+    {
+      code: "InvalidArg",
+      message: "expect String, got: Object"
+    }
+  );
+
+})
+
+
+test('get/set number properties', (t) => {
+
+  let compiler = new ComponentCompiler;
+  let definition = compiler.buildFromSource(`export component App { in-out property <float> age: 42; }`, "");
+  t.not(definition, null);
+
+  let instance = definition!.create();
+  t.not(instance, null);
+
+  t.is(instance!.getProperty("age"), 42);
+
+  instance!.setProperty("age", 100);
+  t.is(instance!.getProperty("age"), 100);
+
+  t.throws(() => {
+    instance!.setProperty("age", "Hello")
+  },
+    {
+      code: "InvalidArg",
+      message: "expect Number, got: String"
+    }
+  );
+
+  t.throws(() => {
+    instance!.setProperty("age", { "blah": "foo" })
+  },
+    {
+      code: "InvalidArg",
+      message: "expect Number, got: Object"
+    }
+  );
+
+})
+
+test('get/set bool properties', (t) => {
+
+  let compiler = new ComponentCompiler;
+  let definition = compiler.buildFromSource(`export component App { in-out property <bool> ready: true; }`, "");
+  t.not(definition, null);
+
+  let instance = definition!.create();
+  t.not(instance, null);
+
+  t.is(instance!.getProperty("ready"), true);
+
+  instance!.setProperty("ready", false);
+  t.is(instance!.getProperty("ready"), false);
+
+  t.throws(() => {
+    instance!.setProperty("ready", "Hello")
+  },
+    {
+      code: "InvalidArg",
+      message: "expect Boolean, got: String"
+    }
+  );
+
+  t.throws(() => {
+    instance!.setProperty("ready", { "blah": "foo" })
+  },
+    {
+      code: "InvalidArg",
+      message: "expect Boolean, got: Object"
+    }
+  );
+
+})
+
+test('set struct properties', (t) => {
+  let compiler = new ComponentCompiler;
+  let definition = compiler.buildFromSource(`
+  export struct Player {
+    name: string,
+    age: int
+  }
+  export component App {
+    in-out property <bool> bool-prop;
+    in-out property <Player> player: {
+      name: "Florian",
+      age: 20,
+    };
+  }
+  `, "");
+  t.not(definition, null);
+
+  let instance = definition!.create();
+  t.not(instance, null);
+
+  t.false(instance!.getProperty("bool-prop"));
+  instance!.setProperty("bool-prop", true);
+  t.true(instance!.getProperty("bool-prop"));
```

Comment:
I'm curious, is there a specific reason for the boolean property test in this struct test?